### DISCUSSION TITLE
working-directory doesn't seem to work

### DISCUSSION THREAD

**Discussion by NetOpWibby** [ghostty-org/ghostty#4123] [Q&A]
I have mine set to `working-directory = /Users/netopwibby/Projects` but when I reload config or close/open the app again, I'm in `~`.

EDIT: I'm on macOS Sequoia 15.2 (24C101), if that helps.

**Comment by rhodes-b** [comment]
This might be related to the detection logic defaulting to HOME as noted here
https://ghostty.org/docs/config/reference#working-directory

(just speculating did not look into the codebase)
I just tested this on Linux and it seemed to work (be sure there are no instances of ghostty still alive when you re launch sometimes it does not quit on close)

**Reply by NetOpWibby** [reply]
I restarted my computer just to be sure and unfortunately, it didn't take.

**Comment by Winterhuman** [comment]
For anyone still running into this issue with Ghostty v1.1.3 or later: Enabling `gtk-single-instance` causes the `working-directory` option to be ignored; all instances use the same working directory as the first instance (even when `window-inherit-working-directory = false` is set). If you're spawning `ghostty` using a file manager's "Open Terminal Here" button, you need to use `ghostty --gtk-single-instance=false` as the command (so you can keep single instancing enabled in the config globally).

Perhaps Ghostty should show a warning if both of these options are used together? Like a pop-up saying "Gtk single instancing enabled, ignoring working directory setting.", a config error, or maybe just a debug print?

**Reply by SinTan1729** [reply]
It might be useful if at least `ghostty` passed down the variables to the startup command. I can handle it from there using my own logic in `tmux`.

**Comment by gernotstarke** [comment]
On MacOS Tahoe 26.2, the open "working-directory" is ignored. Combined with the problem that "-e cd ~...." commands are ignored too, it renders ghostty unusable for me (I have to start a group of terminals each in distinct dirs via script).

**Reply by mludi** [reply]
Any update here?

**Comment by mrsimpson** [comment]
@gernotstarke I ran into a related issue on macOS: Splits always started in home instead of the previous working-directory. What finally worked for me was having zsh emit OSC 7 before each prompt, so Ghostty can track the current working directory:

```
autoload -Uz add-zsh-hook
__emit_osc7_cwd() {
  print -Pn '\e]7;file://%m%d\e\\'
}
add-zsh-hook precmd __emit_osc7_cwd
```

With `shell-integration = detect`, new tabs and splits now inherit the active directory reliably. This was inspired by [this gist](https://gist.github.com/fengyu225/b3f5c4181d98d73385bf1f2065ae517a/9d7a09e02976271a12c115e998e66f102c7fed32), though its example is for bash. It doesn’t solve the launch-time working-directory issue, but hopefully helps as a workaround.
